CHANGELOG.md

```diff
@@ -1,6 +1,20 @@
 CHANGELOG
 =========
 
+### 4.0.0 (2016-xx-xx)
+
+* Added: Interface for `Geocoder\Model\AddressCollection` called `Geocoder\GeocoderResult`. Public APIs are updated to type hint for `Geocoder\GeocoderResult`.
```

Comment:
I'm happy with any name but `AddressCollectionInterface`. Do you have any suggestions? Or do we want the "Interface" suffix?